src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -14,95 +14,95 @@ import { updatePlotConfig } from '../../../redux/actions/componentConfig';
 import ReorderableList from '../../ReorderableList';
 import { getCellSetsHierarchy } from '../../../redux/selectors';
 
+const convertToReorderableListData = (cellClassKeys, selected, hierarchy) => (
+  cellClassKeys.map((key) => {
+    const { name } = _.find(hierarchy, ({ key: currentKey }) => currentKey === key);
+
+    return {
+      key, disabledReorder: !selected, name,
+    };
+  })
+);
+
 const HeatmapMetadataTrackSettings = (props) => {
   const dispatch = useDispatch();
 
   const { componentType } = props;
-  const selectedHierarchy = useSelector(getCellSetsHierarchy());
+  const hierarchy = useSelector(getCellSetsHierarchy());
 
   const selectedTracks = useSelector(
     (state) => state.componentConfig[componentType].config.selectedTracks,
   );
 
-  const getTrackData = () => selectedHierarchy.map(
-    (data) => ({ selected: selectedTracks.includes(data.key), key: data.key }),
-  );
+  const [listData, setListData] = useState([]);
 
-  const isInitialRenderRef = useRef(true);
+  useEffect(() => {
+    if (hierarchy === null || _.isEmpty(hierarchy)) return;
 
-  const [trackData, setTrackData] = useState(getTrackData());
+    const selectedTracksListData = convertToReorderableListData(
+      selectedTracks, true, hierarchy,
+    );
 
-  const getUpdatedTrackData = () => _.unionBy(
-    getTrackData(),
-    trackData,
-    'key',
-  );
+    const unselectedTracks = _.difference(hierarchy.map(({ key }) => key), selectedTracks);
+    const unselectedTracksListData = convertToReorderableListData(
+      unselectedTracks, false, hierarchy,
+    );
 
-  useEffect(() => {
-    // Prevent initial dispatch when object appears
-    if (isInitialRenderRef.current) return;
+    setListData([...selectedTracksListData, ...unselectedTracksListData]);
+  }, [selectedTracks]);
 
-    const newTrackData = getUpdatedTrackData();
-    if (_.isEqual(trackData, newTrackData)) return;
+  const setTrackSelected = (selected, key) => {
+    const newSelectedTracks = [...selectedTracks];
 
-    setTrackData(newTrackData);
-  }, [selectedHierarchy]);
+    if (selected) {
+      newSelectedTracks.push(key);
+    } else {
+      _.pull(newSelectedTracks, key);
+    }
 
-  const getEnabledTracks = () => trackData.filter((entry) => entry.selected).map((o) => o.key);
+    dispatch(
+      updatePlotConfig(componentType, {
+        selectedTracks: newSelectedTracks,
+      }),
+    );
```

Comment:
this triggers the `useEffect` in lines 39-52, is this intended?

Reply:
Resolving this one because it seems like the same topic as https://github.com/biomage-ltd/ui/pull/591#discussion_r773702048